src/main/java/mcp/mobius/betterbarrels/common/items/dolly/ItemBarrelMover.java

```diff
@@ -42,6 +45,7 @@ public class ItemBarrelMover extends Item {
     protected IIcon text_empty = null;
     protected IIcon text_filled = null;
     protected DollyType type = DollyType.NORMAL;
+    protected boolean preventFold = false;
```

Comment:
Remove this field.
Because there is *only one instance* of `ItemBarrelMover` (defined as field in main mod class) then this field is responsible for all dollies of all players on server. In other words, if one player uses dolly then all dollies of this player and all dollies on server will become locked.
If you want to prevent stacking of dolly right after placement of barrel, you should use NBT of `ItemStack`.

Reply:
I've made the changes you've requested. Since the tag is applied and removed in the same tick, I didn't bother to persist it.